R/CohortCharacterizationDiagnostics.R

```diff
@@ -22,176 +22,162 @@ getCohortCharacteristics <- function(connectionDetails = NULL,
-          )
+    covariates <- featureExtractionOutput$covariates %>%
+      dplyr::rename(cohortId = .data$cohortDefinitionId) %>%
+      dplyr::left_join(populationSize, by = "cohortId", copy = TRUE) %>%
+      dplyr::mutate(p = .data$sumValue / .data$populationSize)
+    
+    if (nrow(covariates %>%
+             dplyr::filter(.data$p > 1) %>%
+             dplyr::collect()) > 0) {
+      stop(
+        paste0(
+          "During characterization, population size (denominator) was found to be smaller than features Value (numerator).",
+          "- this may have happened because of an error in Feature generation process. Please contact the package developer."
         )
-      }
-
+      )
+    }
+    
+    covariates <- covariates %>%
+      dplyr::mutate(sd = sqrt(.data$p * (1 - .data$p))) %>%
+      dplyr::select(-.data$p) %>%
+      dplyr::rename(mean = .data$averageValue) %>%
+      dplyr::select(-.data$populationSize)
+    
+    if (FeatureExtraction::isTemporalCovariateData(featureExtractionOutput)) {
       covariates <- covariates %>%
-        dplyr::mutate(sd = sqrt(.data$p * (1 - .data$p))) %>%
-        dplyr::select(-.data$p) %>%
-        dplyr::rename(mean = .data$averageValue) %>%
-        dplyr::select(-.data$populationSize)
-
-      if (FeatureExtraction::isTemporalCovariateData(featureExtractionOutput)) {
-        covariates <- covariates %>%
-          dplyr::select(
-            .data$cohortId,
-            .data$timeId,
-            .data$covariateId,
-            .data$sumValue,
-            .data$mean,
-            .data$sd
-          )
-          if (length(is.na(covariates$timeId)) > 0) {
-            covariates[is.na(covariates$timeId),]$timeId <- -1
-          }
-      } else {
-        covariates <- covariates %>%
-          dplyr::mutate(timeId = 0) %>%
-          dplyr::select(
-            .data$cohortId,
-            .data$timeId,
-            .data$covariateId,
-            .data$sumValue,
-            .data$mean,
-            .data$sd
-          ) %>%
-          dplyr::mutate(timeId = 0)
-      }
-      if ("covariates" %in% names(results)) {
-        Andromeda::appendToTable(results$covariates, covariates)
-      } else {
-        results$covariates <- covariates
+        dplyr::select(
+          .data$cohortId,
+          .data$timeId,
+          .data$covariateId,
+          .data$sumValue,
+          .data$mean,
+          .data$sd
+        )
+      if (length(is.na(covariates$timeId)) > 0) {
+        covariates[is.na(covariates$timeId), ]$timeId <- -1
       }
+    } else {
+      covariates <- covariates %>%
+        dplyr::mutate(timeId = 0) %>%
+        dplyr::select(
+          .data$cohortId,
+          .data$timeId,
+          .data$covariateId,
+          .data$sumValue,
+          .data$mean,
+          .data$sd
+        ) %>%
+        dplyr::mutate(timeId = 0)
     }
-
-    if ("covariatesContinuous" %in% names(featureExtractionOutput) &&
+    if ("covariates" %in% names(results)) {
```

Comment:
Looks like most of these conditional cases aren't tested. Is this because of the features used (e.g. tests before weren't using continuous features) or some other reasons?

Maybe these tests can be improved?

Reply:
I am sure it can be improved. Those were not tested before. This is a refactor makes it possible to test them! But i have added the main tests, not adding any more

Reply:
I am sure it can be improved. Those were not tested before. This is a refactor makes it possible to test them! But i have added the main tests, dont have bandwidth to add more tests. We can do it at a future test a thon thing